src/Interpreters/Context.cpp

```diff
@@ -785,11 +785,12 @@ struct ContextSharedPart : boost::noncopyable
         SHUTDOWN(log, "UDFs loader", external_user_defined_executable_functions_loader, enablePeriodicUpdates(false));
         SHUTDOWN(log, "another UDFs storage", user_defined_sql_objects_storage, stopWatching());
         SHUTDOWN(log, "workload entity storage", workload_entity_storage, stopWatching());
+        SHUTDOWN(log, "ddl worker", ddl_worker, stopWatching());
```

Comment:
It's necessary to stop watching for enqueued tasks in `DDLWorker` before calling `BackupsWorker::shutdown()`, because otherwise it's possible that `DDLWorker` starts executing an internal BACKUP/RESTORE query initiated by a different node after `BackupsWorker` has already shut down.